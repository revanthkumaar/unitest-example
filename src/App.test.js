import React from 'react';
import Enzyme, {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Account from './account-component';
import toJson from 'enzyme-to-json'


configure({adapter: new Adapter})


const user = {
  name: 'revanth',
  email: 'example.com',
  username: 'revdd'
}

describe("testing the render of account component", () => {

/*
      it("accepts user account props", ()=> {
      const wrapper = mount(<Account profile={user}/>);
      expect(wrapper.props().profile).toEqual(user);
    });


    it("contains users account email", () => {
      const wrapper = mount(<Account profile={user}/>);
      const email = wrapper.find("p").text()
      expect(email).toEqual("example.com")
    })
*/
    //snapshot test case:

    it('acccount comp renders correctly', () => {
        const tree = shallow(<Account/>);
        expect(toJson(tree)).toMatchSnapshot();
    })


})



