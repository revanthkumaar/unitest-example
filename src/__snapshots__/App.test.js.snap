// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the render of account component acccount comp renders correctly 1`] = `
<div
  className="Account"
>
  <p>
    revnht
  </p>
</div>
`;
